communication-core/sockets/mock/mock_udp_multicast_socket.h

```diff
@@ -0,0 +1,26 @@
+/**
+ * @file mock_udp_multicast_socket.h
+ * @author Jacek Kukawski (jacekka6@gmail.com)
+ * @brief a mock for udp_multicast_socket.h
+ * @version 1.0
+ * @date 2024-06-17
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+#ifndef COMMUNICATION_CORE_SOCKETS_MOCK_MOCK_UDP_MULTICAST_SOCKET_H_
+#define COMMUNICATION_CORE_SOCKETS_MOCK_MOCK_UDP_MULTICAST_SOCKET_H_
+#include "gmock/gmock.h"
+#include "udp_multicast_socket.h"
+
+
+class MockUDPMulticastSocket : public simba::com::soc::UDPMulticastSocket {
```

Comment:
to powinno dziedziczyć po interfejsie ISocket bodajże, tak nie zawsze będzie działać